### PR TITLE
[Inet] Add Openthread Native ipv6 address

### PR DESCRIPTION
**Pull request by jepenven-silabs**
#### Problem
LwIP is used as a wrapper inside the Inet layer. This cause bloating for Native OpenThread implementation of Matter since there is a duplications of API and libs.

#### Change overview
Add OpenThread native IPV6 address api in IPAddress.cpp
Add InterfaceIterator implementation for platform with only 1 OpenThread interface.

#### Testing
Tested with EFR32 and Unit Test

### CODE REVIEW

File: src/inet/InetInterface.cpp [hunk 1]
```diff
@@ -63,7 +63,81 @@
...
+
+    return CHIP_NO_ERROR;
+}
+CHIP_ERROR InterfaceId::InterfaceNameToId(const char * intfName, InterfaceId & interface)
+{
+    if (strlen(intfName) < 3)
+    {
+        return INET_ERROR_UNKNOWN_INTERFACE;
+    }
+    char * parseEnd;
```
Comment (tcarmelveilleux) [inline src/inet/InetInterface.cpp:88]: set to nullptr by default

File: src/inet/InetInterface.cpp [hunk 2]
```diff
@@ -63,7 +63,81 @@
...
+    return CHIP_NO_ERROR;
+}
+
+uint8_t InterfaceAddressIterator::GetPrefixLength()
+{
+    if (HasCurrent())
+    {
+        return 64;
+    }
+    return 0;
```
Comment (tcarmelveilleux) [inline src/inet/InetInterface.cpp:135]: Zero is never valid. If only 64 bit prefixes are supported, consider returning 64 always

**Review by tcarmelveilleux — APPROVED** [review]

File: src/inet/IPAddress.cpp [hunk 3]
```diff
@@ -269,6 +269,26 @@ CHIP_ERROR IPAddress::GetIPAddressFromSockAddr(const SockAddr & sockaddr, IPAddr
 
 #endif // CHIP_SYSTEM_CONFIG_USE_SOCKETS || CHIP_SYSTEM_CONFIG_USE_NETWORK_FRAMEWORK
 
+#if CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT
+IPAddress::IPAddress(const otIp6Address & ipv6Addr)
+{
+    memcpy(Addr, ipv6Addr.mFields.m32, sizeof(Addr));
```
Comment (bzbarsky-apple) [inline src/inet/IPAddress.cpp:275]: Maybe we should static_assert somewhere here that sizeof(Addr) == sizeof(ipv6Addr.mFields.m32)?

File: src/inet/arpa-inet-compatibility.h [hunk 4]
```diff
@@ -46,4 +46,40 @@
...
+#ifndef htonl
+#define htonl(x) (x)
+#endif
+#ifndef ntohl
+#define ntohl(x) (x)
+#endif
+
+#else // BYTE_ORDER != BIG_ENDIAN
+#ifndef htons
+#define htons(x) ((u16_t)((((x) & (u16_t) 0x00ffU) << 8) | (((x) & (u16_t) 0xff00U) >> 8)))
```
Comment (bzbarsky-apple) [inline src/inet/arpa-inet-compatibility.h:68]: Wonder if we could use our existing byteswap functions here... but seems fine.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/inet/InetInterface.cpp (hunk 1 above)
Comment (jepenven-silabs) [reply src/inet/InetInterface.cpp:88]: fixed

File: src/inet/InetInterface.cpp (hunk 2 above)
Comment (jepenven-silabs) [reply src/inet/InetInterface.cpp:135]: fixed

File: src/inet/IPAddress.cpp (hunk 3 above)
Comment (jepenven-silabs) [reply src/inet/IPAddress.cpp:276]: Added some static_assert

File: src/inet/arpa-inet-compatibility.h (hunk 4 above)
Comment (jepenven-silabs) [reply src/inet/arpa-inet-compatibility.h:68]: leaving as is for now

**Pull request merged**